js/core/markdown.js

```diff
@@ -1,280 +1,232 @@
+      addSection: addSection,
+      addElement: addElement,
+      getRoot: getRoot
+    };
+  }
+
+  function structure(fragment, doc) {
+    function process(root) {
+      var node;
+      var tagName;
+      var stack = makeBuilder(doc);
+
+      while (root.firstChild) {
+        node = root.firstChild;
+        if (node.nodeType !== 1) {
+          root.removeChild(node);
+          continue;
+        }
+        tagName = node.localName;
+        switch (tagName) {
+        case 'h1':
+        case 'h2':
+        case 'h3':
+        case 'h4':
+        case 'h5':
+        case 'h6':
+          stack.addHeader(node);
+          break;
+        case 'section':
+          stack.addSection(node, process);
+          break;
+        default:
+          stack.addElement(node);
+        }
+      }
 
-                return process(fragment);
-            },
+      return stack.getRoot();
+    }
 
-            run: function (conf, doc, cb, msg) {
-                msg.pub("start", "core/markdown");
-                if (conf.format === 'markdown') {
-                    // Marked, the Markdown implementation we're currently using
-                    // parses markdown nested in markup (unless it's in a section element).
-                    // Turns out this is both what we need and generally not what other
-                    // parsers do.
-                    // In case we switch to another parser later on, we'll need to
-                    // uncomment the below line of code.
-                    //
-                    // this.processIssuesNotesAndReqs(doc);
-                    this.processSections(doc);
-                    // the processing done here blows away the ReSpec UI (or rather, the elements
-                    // that it needs to reference). So we save a reference to the original element
-                    // and re-inject it later
-                    var $rsUI = $("#respec-ui");
-                    var fragment = this.structure(this.processBody(doc), doc);
-                    doc.body.innerHTML = '';
-                    doc.body.appendChild(fragment);
-                    if ($rsUI.length) $("#respec-ui").replaceWith($rsUI);
-                }
-                msg.pub("end", "core/markdown");
-                cb();
-            }
-        };
+    return process(fragment);
+  }
+  var processSections = processElements("section");
+  var processIssuesNotesAndReqs = processElements(".issue, .note, .req");
+
+  return {
+    run: function (conf, doc, cb, msg) {
+      msg.pub("start", "core/markdown");
+      if (conf.format === 'markdown') {
+        // We transplant the UI to do the markdown processing
+        var rsUI = doc.getElementById("respec-ui");
+        rsUI.remove();
+        // Marked expects markdown be flush against the left margin
+        // so we need to normalize the inner text of all block
+        // elements.
+        processSections(doc);
+        processIssuesNotesAndReqs(doc);
```

Comment:
Curious as to why this suddenly needs to become uncommented. Has marked stopped parsing markdown in nested markup?